source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -124,6 +126,25 @@ const mem_stat_t *ns_dyn_mem_get_mem_stat(void)
 #endif
 }
 
+int ns_dyn_mem_read_mem_stat(mem_gen_stat_t *mem_gen_stat)
+{
+#ifndef STANDARD_MALLOC
+    if (!default_book) {
+        return -1;
+    }
+
+    platform_enter_critical();
+    mem_gen_stat->heap_sector_size = default_book->heap_size;
+    mem_gen_stat->heap_alloc_fail_cnt = default_book->heap_alloc_fail_cnt;
+    mem_gen_stat->heap_alloc_total_bytes = default_book->heap_alloc_total_bytes;
```

Comment:
NULL check is missing from parameter mem_gen_stat.

The signature of this API is:
` int ns_dyn_mem_read_mem_stat(mem_gen_stat_t *mem_gen_stat)`
And the other related API is:
` const mem_stat_t *ns_dyn_mem_get_mem_stat(void)`
Should the API naming be aligned? 

Now we are having two methods for the same purpose. Why we need two?